### PR TITLE
Login and signup

### PR DESCRIPTION
**Pull request by smithpg**
Implemented signup / login page, not including integration with backend and handling of backend response. In the process I made a Button and Navbar component which we will be able to use across the app.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -14,6 +14,8 @@ node_modules/
 .env.test.local
 .env.production.local
 
+config/
```
Comment (skassam21) [inline .gitignore:17]: I don't know if you'd want this whole folder to be gitignore? Maybe you use environment variables in config folder, and then ignore .env

File: client/src/components/Button.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+import styled from "styled-components";
+
+import { colors } from "../themes/theme";
+const { brand, brandLight } = colors;
+
+const Button = styled.button`
+  padding: 15px;
+  color: white;
+  background: ${brandLight};
+  font-family: "Montserrat";
```
Comment (skassam21) [inline client/src/components/Button.js:11]: Would this be handled in the theme?

File: client/src/App.js [hunk 3]
```diff
@@ -1,19 +1,28 @@
...
 import "./App.css";
 
+//TODO: Implement this
+function browserHasToken() {
+  return false;
+}
+
 function App() {
   return (
-    <MuiThemeProvider theme={theme}>
```
Comment (skassam21) [inline client/src/App.js:12]: are you planning on not using themes at all?

File: client/public/index.html [hunk 4]
```diff
@@ -16,6 +16,10 @@
       type="text/css"
       href="https://fonts.googleapis.com/css?family=Roboto:300,400,500,700|Material+Icons"
```
Comment (skassam21) [inline client/public/index.html]: can probably remove Roboto and keep Material+Icons, if you are not using Roboto anymore

File: client/src/components/LogInForm.js [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+  function onChange(e) {
+    const {
+      target: { name, value }
+    } = e;
+
+    setFormValues({
+      [name]: value
+    });
+  }
```
Comment (skassam21) [inline client/src/components/LogInForm.js:20]: is onChange used?

File: client/src/components/LogInForm.js [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+
+  return (
+    <form onSubmit={onSubmit}>
+      <TextField
+        label={"Email"}
+        value={email}
+        name="email"
+        onChange={e => {
+          e.preventDefault();
```
Comment (skassam21) [inline client/src/components/LogInForm.js]: don't need the preventDefaults for onChange

File: client/src/components/LogInForm.js [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+
+      <TextField
+        label={"Password"}
+        value={password}
+        onChange={e => {
+          e.preventDefault();
+          setFormValues({ password: e.target.value });
+        }}
+      />
+
```
Comment (lei29) [inline client/src/components/LogInForm.js:39]: May forgot the link of forgot your password.

Comment (smithpg) [reply client/src/components/LogInForm.js:39]: Good catch, I'll add that in and maybe we can punt for now on the issue of figuring out how to support password recovery on the backend

File: client/src/App.js (hunk 3 above)
Comment (smithpg) [reply client/src/App.js:12]: Yeah, I think we can probably do without using a theme provider like this. We can still use the theme.js file as a central record for style related constants like colors, typography, rhythm, etc.

File: .gitignore (hunk 1 above)
Comment (smithpg) [reply .gitignore:17]: My mistake, I was under the impression that config was basically a way to do .env in .json. But I can see how there would be plenty of non-secret pieces of config data that would be worth comitting.

**Review by lei29 — APPROVED** [review]

**Pull request merged**